core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -314,7 +323,7 @@ private <T> T execute(
 
   @Override
   public void head(String path, Map<String, String> headers, Consumer<ErrorResponse> errorHandler) {
-    execute(Method.HEAD, path, null, null, null, headers, errorHandler);
+    execute(Method.HEAD, buildBaseUri(path), null, null, null, headers, errorHandler);
```

Comment:
I don't follow why we're updating all of these methods since the all delegate to the execute method.  This feels unnecessary and should just be handled in the execute method.